examples/rl/ddpg_pendulum.py

```diff
@@ -209,23 +198,29 @@ def learn(self):
             zip(actor_grad, actor_model.trainable_variables)
         )
 
+    # We compute the loss and update parameters
+    def learn(self):
+        # Get sampling range
+        record_range = min(self.buffer_counter, self.buffer_capacity)
+        # Randomly sample indices
+        batch_indices = np.random.choice(record_range, self.batch_size)
 
-# This update target parameters slowly
-# Based on rate `tau`, which is much less than one.
-def update_target(tau):
-    new_weights = []
-    target_variables = target_critic.weights
-    for i, variable in enumerate(critic_model.weights):
-        new_weights.append(variable * tau + target_variables[i] * (1 - tau))
+        # Convert to tensors
+        state_batch = tf.convert_to_tensor(self.state_buffer[batch_indices])
+        action_batch = tf.convert_to_tensor(self.action_buffer[batch_indices])
+        reward_batch = tf.convert_to_tensor(self.reward_buffer[batch_indices])
+        reward_batch = tf.cast(reward_batch, dtype=tf.float32)
+        next_state_batch = tf.convert_to_tensor(self.next_state_buffer[batch_indices])
 
-    target_critic.set_weights(new_weights)
+        self.update(state_batch, action_batch, reward_batch, next_state_batch)
 
-    new_weights = []
-    target_variables = target_actor.weights
-    for i, variable in enumerate(actor_model.weights):
-        new_weights.append(variable * tau + target_variables[i] * (1 - tau))
 
-    target_actor.set_weights(new_weights)
+# This update target parameters slowly
+# Based on rate `tau`, which is much less than one.
+@tf.function
+def update_target(target_weights, weights, tau):
+    for (a, b) in zip(target_weights, weights):
+        tf.compat.v1.assign(a, b * tau + a * (1 - tau))
```

Comment:
The way to assign a value to a variable is via `var.assign(value)`.

Reply:
Fixed!